Add example for real-time profiles (#25)
bakdata/quick: docs/docs/user/examples/real-time-monitoring.md

```

#### Topics
Next, we create all required topics.
The command expects the topic name as well as the type or schema of key and value.
Since we have complex values, we reference the GraphQL types.

```shell
quick topic create vehicle -k string -v schema --schema car-sharing.Vehicle
quick topic create status -k string -v schema --schema car-sharing.Status
quick topic create trip -k string -v schema --schema car-sharing.Trip
```

#### Application
Then, we can start our Kafka Streams application.
Quick supports running dockerized applications.

```shell
quick app deploy trip-aggregator \
 --registry bakdata \
 --image quick-demo-monitoring-trip-aggregator \
 --tag latest \
 --args input-topics=status output-topic=trip
```

## Go Live

When all resources are up, we can start to ingest data into our system.
Quick supports the ingest through a REST-API.
For example, the following snippet shows a command ingesting new vehicles into the `vehicle` topic.

```shell
curl -X POST --url $QUICK_URL/ingest/vehicle \
  --header "content-type: application/json" \
  --header "X-API-Key:$QUICK_API_KEY" \
  --data "@./simulator/data/vehicles.json"
```

> Exemplary data are contained in the [simulator subdirectory](https://github.com/bakdata/quick-examples/tree/main/carsharing/simulator).

Reviewer: should work after https://github.com/bakdata/quick-examples/pull/8

# Real-time monitoring and analytics

> With this use case, we demonstrate how Quick can be used to process data streams

Author: Could you rewrite this file a little to reduce the usage of "we"? See https://docs.microsoft.com/en-us/style-guide/grammar/person#avoid-plural-first-person-we-us

We simply group the incoming status events by their trip id and append them to a list.
The result is written into the trip topic.

```java
void buildTopology(StreamsBuilder builder){
    builder.stream("status")
        .groupBy((key,status)->status.getTripId())
        .aggregate(Trip::new,this::aggregateTrip)
        .toStream()
        .to("trip");
}
```

```java
Trip aggregateTrip(String tripId, Status newStatus, Trip trip){
    List<Status> route = trip.getRoute();
    // first time we see this trip id
    if (route == null) {
        trip.setId(tripId);
        trip.setVehicleId(newStatus.getVehicleId());
        route = new ArrayList<>();
        trip.setRoute(route);
    }

    route.add(newStatus);
    return trip;
}
```

You can find the full code in our [example repository](https://github.com/bakdata/quick-examples/tree/main/carsharing).
The Kafka Streams application is written with
our [streams-bootstrap library](https://github.com/bakdata/streams-bootstrap),
which, among others, offers sensible defaults and reduces the required boilerplate code.
Of course, you are not limited to simple aggregation.
With Kafka Streams you can also build more advanced applications.

> One could for example build a predictive maintenance service with it.

Reviewer 2: I don't like this part of the text. I think the user reading this already knows the capabilities of Kafka Streams. I would remove this.

## GraphQL schema

> Having our topics defined, we start by modeling the data required in the dashboard.

Reviewer 2: ```suggestion
After defining the topics, it is time to model the data required in the dashboard.
```

Quick’s querying logic is built upon the data query language GraphQL.
It allows us to create a global schema of our data and the supported operations.
Subscriptions are one type of such operations, allowing us to consume real-time data updates of the data through
WebSocket connections.
This is an exemplary GraphQL schema for live updates of the emitted status events.

> With that, we have a subscription operation called `statusUpdates` that we can use to get live updates of `Status`

Reviewer 2: As @torbsto pointed out you can consider rephrasing the sentences with we, our.

events.
```graphql
type Subscription {
    statusUpdates: Status @topic(name: "status")
}
```

> The events have the following schema.

Reviewer 2: ```suggestion
The status events have the following schema.
```

    statusUpdates: Status @topic(name: "status")
}
```
Second, we want to display information about a vehicle when querying a trip.
Instead of creating a separate operation, we can add this information to `Trip` itself:
`Trip` has a new field `vehicle`.
It is populated with the `vehicle` topic data based on the trip’s `vehicleId` value.
One major advantage of GraphQL is its flexibility.
When querying a trip, the user can decide if they indeed require the vehicle information.
If this is not the case, the corresponding data is never loaded, and thus no overhead occurs.

```graphql
type Query {
    trip(id: String): Trip @topic(name: "trip", keyArgument: "id")
}

type Trip {
    id: String!,
    vehicleId: String!,
    vehicle: Vehicle @topic(name: "vehicle", keyField: "vehicleId")
    route: [Status]
}

type Vehicle {
    id: String!,
    name: String!,
    maxRange: Int!
}
```

## Quick

We are ready to process and query our data with Quick.
To start a Quick instance, you can refer to the [getting started guide](../../getting-started/setup-quick).
If you haven't done so already, you need to create a Quick context with the CLI.
```shell
quick context create --host $HOST --key $KEY

> ```

Reviewer 2: Still not a big fan of this. I would just refer the user to the (../../getting-started/setup-cli)
It is pretty misleading having the CLI command there.
